tt_metal/common/test_tiles.hpp

```diff
@@ -65,20 +76,31 @@ std::vector<T> convert_to_tile_layout(const BufferType<T>& data) {
 }
 
 template <class T, template <typename...> typename BufferTyp>
-std::vector<T> convert_to_flat_layout(const BufferTyp<T>& data) {
+std::vector<T> convert_to_flat_layout(
+    const BufferTyp<T>& data,
+    const std::optional<std::vector<uint32_t>>& tile_shape = std::nullopt,
+    const std::optional<const std::vector<uint32_t>>& face_shape = std::nullopt) {
     ZoneScoped;
     std::vector<T> result;
```

Comment:
Same comment as above, here it is more important as you are pushing back individual elements one at a time.

Reply:
changed